JavaScript/JavaScriptSDK/Context/User.ts

```diff
@@ -50,10 +97,38 @@ module Microsoft.ApplicationInsights.Context {
                 // 365 * 24 * 60 * 60 * 1000 = 31536000000 
                 date.setTime(date.getTime() + 31536000000);
                 var newCookie = [this.id, acqStr];
-                Util.setCookie('ai_user', newCookie.join('|') + ';expires=' + date.toUTCString());
+                Util.setCookie(User.userCookieName, newCookie.join(User.cookieSeparator) + ';expires=' + date.toUTCString());
             }
 
+            // We still take the account id from the ctor param for backward compatibility. 
+            // But if the the customer set the accountId through the newer setAuthUserContext API, we will override it.
             this.accountId = accountId;
+
+            // Get the auth user id and account id from the cookie if exists
+            // Cookie is in the pattern: <authId>|<accountId>
+            var authCookie = Util.getCookie(User.authUserCookieName);
+            if (authCookie) {
+                authCookie = decodeURI(authCookie);
+                var authCookieString = authCookie.split(User.cookieSeparator);
+                if (authCookieString[0]) {
+                    this.authId = authCookieString[0];
+                }
+                if (authCookieString.length > 1 && authCookieString[1]) {
+                    this.accountId = authCookieString[1];
+                }
+            }
+        }
+
+        private validateUserInput(id: string) {
+            // Validate:
+            // 1. Id is a non-empty string.
+            // 2. It does not contain special characters for cookies.
+            if (typeof id !== 'string' ||
+                !id ||
+                id.match(/,|;|=| |\|/)) {
+                throw new Error("User auth id and account id should be of type string. They should not contain commas, semi-colons, equal signs, spaces, or vertical-bars.");
```

Comment:
I originally wrote this as replace, but the PM's didnt like the idea of manipulating the user input - the user might search for the exact string later in the UI and not find it.

If we're rejecting bad input, then we have to log it somehow - just ignoring it is not a good experience IMO.
Would you prefer this to be:
_InternalLogging.throwInternalUserActionable(LoggingSeverity.WARNING, "Setting auth user context failed..... ");
?